diff --git a/drivers/mmc/host/sdhci-pic32.c b/drivers/mmc/host/sdhci-pic32.c
index ac1f8a9e7..e6db4832f 100644
--- a/drivers/mmc/host/sdhci-pic32.c
+++ b/drivers/mmc/host/sdhci-pic32.c
@@ -86,14 +86,7 @@ void pic32_sdhci_set_bus_width(struct sdhci_host *host, int width)
 		else
 			ctrl &= ~SDHCI_CTRL_4BITBUS;
 	}
-	/*
-	 * SDHCI will not work if JTAG is not Connected. As a workaround fix,
-	 * set Card Detect Signal Selection bit in SDHCI Host Control
-	 * register and clear Card Detect Test Level bit in SDHCI Host
-	 * Control register.
-	 */
-	ctrl &= ~SDHCI_CTRL_CDTLVL;
-	ctrl |= SDHCI_CTRL_CDSSEL;
+
 	sdhci_writeb(host, ctrl, SDHCI_HOST_CONTROL);
 }
 
